### PR TITLE
Event logging docs

### PR DESCRIPTION
**Pull request by Kargathia**
Spec document for how history service should function

### CODE REVIEW

File: tech_specs/event_logging.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+In order to decouple individual controllers and the history database, logging is done through events.
+Controllers publish events with their latest state, without knowing who reads it.
+
+The history service subscribes to these events, and writes the raw data to the database in a sensible manner.
+
+This way the controller services are not aware of the history service implementation, and the history implementation is not aware of controller service implementation details.
+
+## Data Structure
+
+After a subscription is set, it will relay all incoming messages.
```
Comment (elcojacobs) [inline tech_specs/event_logging.md]: You should probably describe relaying first. Who relays to what?

File: tech_specs/event_logging.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+When relaying, the data dict is flattened.
+The first part of the routing key is considered the controller name, and becomes the InfluxDB measurement name.
```
Comment (elcojacobs) [inline tech_specs/event_logging.md]: I think we should be even more generic than controller here. Service or data source. Not all sources of data control something.

File: tech_specs/event_logging.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        'block1/sensor1/values/other': 1
+    }
+
+If the event data is not a dict, but a string, it is first converted to:
+
+    {
+        'text': <string data>
+    }
+
+This dict is then flattened.
```
Comment (elcojacobs) [inline tech_specs/event_logging.md]: I think this is described in more detail in the actual service. Perhaps just link to that?

Comment (Kargathia) [reply tech_specs/event_logging.md]: It is copy-pasted from the docstring.

File: tech_specs/event_logging.md (hunk 1 above)
Comment (Kargathia) [reply tech_specs/event_logging.md]: Added note that it's described under the previous header.

File: tech_specs/event_logging.md (hunk 2 above)
Comment (Kargathia) [reply tech_specs/event_logging.md]: Fixed

**Pull request merged**